client/src/base/model/smodel.ts

```diff
@@ -194,6 +213,16 @@ export class SModelIndex<E extends SModelElementSchema> {
         return this.id2element.get(id)
     }
 
+    changeId(element: E, newId: string) {
```

Comment:
Not sure whether I like changing IDs. Would it be hard to switch to a "remove and add new" strategy?

Reply:
No, but the performance is better with this new method. I am recomputing ids for routing handles in order to avoid duplicate ids when new handles are added.